src/lib/transition_frontier_controller_tests/test_transition_frontier.ml

```diff
@@ -26,6 +27,42 @@ let%test_module "Root_history and Transition_frontier" =
     let breadcrumb_trail_equals =
       List.equal ~equal:Transition_frontier.Breadcrumb.equal
 
+    let%test "Common ancestor should return the youngest ancestor of two \
+              breadcrumbs" =
+      let logger = Logger.null () in
+      Async.Thread_safe.block_on_async_exn (fun () ->
+          let%bind frontier = create_root_frontier ~logger in
+          let root = Transition_frontier.root frontier in
+          let%bind ancestors =
+            create_breadcrumbs ~logger
+              ~size:(max_length - (max_length / 2))
+              root
+          in
+          let youngest_ancestor = List.last_exn ancestors in
+          let%bind () =
+            Deferred.List.iter ancestors ~f:(fun ancestor ->
+                Transition_frontier.add_breadcrumb_exn frontier ancestor )
+          in
+          let%bind branch1 =
+            create_breadcrumbs ~logger
+              ~size:(1 + Random.int ((max_length / 2) - 1))
```

Comment:
Use QuickCheck to make this try a bunch of possibilities every time you run the test, rather than picking one each time. Also I think the probability of randomly getting two paths of the same length is very low, so add a test that explicitly checks that situation, since that's where the bug was.

Reply:
It's quite slow to generate breadcrumbs. As a result, we only generate one instance of it. I think it would be a good idea to have a test where the forks have even lengths and where the forks don't have even lengths.